server/xcode/xcode_darwin.go

```diff
@@ -126,13 +140,51 @@ func (x *xcodeLocator) locate() error {
 				continue
 			}
 			log.Debugf("Mapped Xcode Version %s=>%s with SDKs %+v", version, developerDirPath, sdks)
-			versionMap[version] = &xcodeVersion{
+			xv := &xcodeVersion{
 				version:          mostPreciseVersion,
 				developerDirPath: developerDirPath,
 				sdks:             sdks,
 			}
+			versionMap[version] = xv
+			if defaultDeveloperDir == developerDirPath {
+				versionMap[""] = xv
```

Comment:
It's kind of unusual to store something under the empty string like this. This would be more idiomatic / clearer if we stored it under a const known key. Something like:

const defaultVersion = "default-xcode-version"

if defaultDeveloperDir == developerDirPath {
				versionMap[defaultVersion] = xv
}

etc

Reply:
Yeah, this was relying on the side effect of if there wasn't an override, it would be the empty string. I'm all for making it explicit.